src/time-picker/time-picker.jsx

```diff
@@ -103,50 +109,76 @@ const TimePicker = React.createClass({
 
   getInitialState() {
     return {
-      time: this.props.defaultTime || emptyTime,
+      time: this._isControlled() ? this._getControlledTime() : this.props.defaultTime,
       dialogTime: new Date(),
       muiTheme: this.context.muiTheme || getMuiTheme(),
     };
   },
 
+  componentWillReceiveProps(nextProps, nextContext) {
+    const newState = this.state;
+    if (nextContext.muiTheme) {
+      newState.muiTheme = nextContext.muiTheme;
+    }
+    if (this._isControlled()) {
+      let newTime = this._getControlledTime(nextProps);
+      if (!DateTime.isEqualTime(newTime, this.state.time)) {
+        newState.time = newTime;
+      }
+    }
+    this.setState(newState);
+  },
+
   windowListeners: {
     'keyup': '_handleWindowKeyUp',
   },
 
-  formatTime(date) {
-    let hours = date.getHours();
-    let mins = date.getMinutes().toString();
-
-    if (this.props.format === 'ampm') {
-      let isAM = hours < 12;
-      hours = hours % 12;
-      let additional = isAM ? ' am' : ' pm';
-      hours = (hours || 12).toString();
-
-      if (mins.length < 2 ) mins = '0' + mins;
-
-      if (this.props.pedantic) {
-        // Treat midday/midnight specially http://www.nist.gov/pml/div688/times.cfm
-        if (hours === '12' && mins === '00') {
-          return additional === ' pm' ? '12 noon' : '12 midnight';
+  formatTime(date, format = 'ampm', pedantic = false) {
+    let result = '';
+    if (date) {
+      let hours = date.getHours();
+      let mins = date.getMinutes().toString();
+
+      if (format === 'ampm') {
+        let isAM = hours < 12;
+        hours = hours % 12;
+        let additional = isAM ? ' am' : ' pm';
+        hours = (hours || 12).toString();
+
+        if (mins.length < 2 ) mins = '0' + mins;
+
+        if (pedantic) {
+          // Treat midday/midnight specially http://www.nist.gov/pml/div688/times.cfm
+          if (hours === '12' && mins === '00') {
+            return additional === ' pm' ? '12 noon' : '12 midnight';
+          }
         }
-      }
 
-      return hours + (mins === '00' ? '' : ':' + mins) + additional;
-    }
+        return hours + (mins === '00' ? '' : ':' + mins) + additional;
+      }
 
-    hours = hours.toString();
+      hours = hours.toString();
 
-    if (hours.length < 2) hours = '0' + hours;
-    if (mins.length < 2) mins = '0' + mins;
+      if (hours.length < 2) hours = '0' + hours;
+      if (mins.length < 2) mins = '0' + mins;
 
-    return hours + ':' + mins;
+      result = hours + ':' + mins;
+    }
+    return result;
   },
 
+  /**
+   * Deprecated.
+   * returns timepicker value.
+   **/
   getTime() {
     return this.state.time;
   },
 
+  /**
+   * Deprecated
+   * sets timepicker value.
+   **/
   setTime(time) {
     this.setState({time: time ? time : emptyTime});
```

Comment:
What about using something like `warning(false, 'setTime is deprecated... ');` ?